### ISSUE TITLE
S/PDIF bitstreaming not working

### ISSUE THREAD

**Issue by Gameboi64** [clsid2/mpc-hc#2131]
Hey, I've been using MPC-HC for a while now as a part of the K-Lite Codec Pack and recently I've been trying to do some S/PDIF bitstreaming, but no matter what I do, I never get a proper bitstreamed AC3/DTS output. This was working last year, as I remember watching a few movies in proper bitstreamed surround with this method, but now it's just not properly working anymore. I've tried absolutely everything, I've looked up all kinds of different issues, nothing whatsoever has helped. I'm not sure if this is a problem with how I'm doing things, but I have used the internal Windows audio settings to do a test for AC3/DTS, and everything does work right then, including the light on my receiver lighting up.

I hope these screenshots can help.
![1](https://github.com/clsid2/mpc-hc/assets/13713250/ebf5c7e7-b92c-4d73-aad3-6dfb4367dc03)
![2](https://github.com/clsid2/mpc-hc/assets/13713250/c8502dc1-3df8-41fc-8adf-3eec4940c54f)
![3](https://github.com/clsid2/mpc-hc/assets/13713250/9802036c-991d-4043-b7a6-521e59535b84)

**Comment by adipose** [comment]
So I would start with identifying which MPC-HC build was last working.  That could be very useful.

If you can't find one, that might mean something else changed on your system.

https://github.com/clsid2/mpc-hc/releases

First, go back as far as you need to to find a working version.  You could guess 12 months based on what you said above.

Then a binary search  (go halfway from the working version to now, if that is working, you go halfway forward, if it's not, you go half way back).  Within 3-4 tests you will have a pretty good idea exactly which build broke it.

**Comment by Gameboi64** [comment]
Well, I went back to the release bundled in with the 1720 build of the K-Lite pack, and it still wasn't working (as I remembered when I watched those movies last year, that was the version I was running).

Turns out after a last-ditch effort failed, I remembered about how these things need exclusive mode. I went to check my audio settings and sure enough I've got a program running on _all_ of my audio devices, so I couldn't lock it into any bitstreamed format. The program does work, unfortunately this has all been user error.

Thank you for your patience, I've been trying to work with this for a couple of hours and only now did it click with me that this may be the issue.

![image](https://github.com/clsid2/mpc-hc/assets/13713250/96b4d3db-0ce4-4df0-9983-e627b6e3dba5)
